### PR TITLE
build: Port `micromamba-feedstock` flags in `micromamba`'s `CMakeLists.txt`

### PR DESCRIPTION
**Pull request by jjerphan**
See [the flags on the feedstock](https://github.com/conda-forge/micromamba-feedstock/blob/898f0787101ef08738bf411b82a5b1baad56e801/recipe/build.sh#L3-L14).

### CODE REVIEW

File: micromamba/CMakeLists.txt [hunk 1]
```diff
@@ -57,6 +57,28 @@ find_package(Threads REQUIRED)
...
+
+    # Dependency of libcurl-static missing in CMakeLists.txt
+    set(
+        frameworks_flags
+        "-framework CoreFoundation -framework CoreServices -framework Security -framework Kerberos"
+    )
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${frameworks_flags}")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${frameworks_flags}")
+endif()
+
```
[the remark this answers is not in the document]
Comment (jjerphan) [reply micromamba/CMakeLists.txt:81]: Could you continue this PR and iterate on a solution?

[the remark this answers is not in the document]
Comment (JohanMabille) [reply micromamba/CMakeLists.txt:81]: BTW you should prefer `target_compile_options` to global cmake flags. By setting them as public on the target libmamba, they will automagically propagate to the target linking with libmamba.

**Comment by jjerphan** [comment]
Closing since this is irrelevant.

File: micromamba/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (henryiii) [reply micromamba/CMakeLists.txt:81]: Also, you should use [`find_library`](https://cmake.org/cmake/help/latest/command/find_library.html). It has special handling for frameworks.

**Pull request closed without merging**